Update Quickstart Documentation for Packaged Applications

kb1lqc opened the pull request:
Faraday Software has undergone some major improvements. Now you can `pip install faraday` or install it from the git repository in 'editable' mode for easy development. This however did break the documentation. This PR focuses on bringing the quickstart tutorials up to date.

Still needs some TLC from me but I'm pushing this now to get some other eyes on the content I have updated. Namely up until 'hello world'

@kb1lqd  @jbemenheiser  @el-iso  @reillyeon  @morrme

kb1lqc committed:
Updated main Start Readme.md

Updated a few grammatical errors and also found a misnumbered power
connector reference!

kb1lqc committed:
Updated Installation Readme documentation

Updated the Faraday quickstart software installation markdown file. This
now includes the pip installation from PyPI and as an editable git
checkout. All zip folder references were removed. Some formatting and
wording of older information was updated as needed.

kb1lqc committed:
Updated Connecting Hardware Documentation

Updated the formatting of this documentation page.

kb1lqc committed:
Initial update to proxy quickstart

Removed a bunch of unecessary info that honestly just belongs in the
normal Proxy documentation. This page was updated with new images and text
regaurding the pip installed version of Faraday. No longer do we suggest
manual INI file editing.

kb1lqc committed:
Removed old unused images

Old proxy images that are no longer relevant were deleted.

kb1lqc committed:
Updated Faraday Configuration Page

Revamped the programming faraday page with new packaged software. This
includes faraday-deviceconfiguration and faraday-simpleconfig.

kb1lqc committed:
Updated Telemetry documentation

Updated quickstart telemetry guide for faraday-telemetry use.

kb1lqc committed:
Merge branch 'master' into issue220

kb1lqc commented:
Yes, Configuring Faraday page needs to be updated since the program options and functionality changed so much today after I wrote the original content for this PR.

kb1lqc committed:
Updated hello world uart tutorial

Updated slightly for better wording + newer import code.

kb1lqc committed:
Corrected callsign-nodeid with REPLACEME

Also added some helpful comments.

kb1lqc committed:
Added link to raw file

Linking to raw file so one can download it if they didn't check out
the project on Github.

kb1lqc committed:
Fixed Pytest Warnings and doc wording

Fixed all Pytest warnings and updated the documentation and code for an
integer instead of README in the node_id variable.

kb1lqc committed:
Added missing Proxy images

Forgot to add these images.

kb1lqc committed:
Updated RF Configuration

Now using the new packaged software and a streamlined configuration guide!

kb1lqc committed:
Updated rfplayground documentation

Minor updates to clarify this page.

kb1lqc committed:
Updated configuration for new Master and images

Added new images for the documentation as well as removed the old leading
--f prefixes for Faraday configuration settings.

kb1lqc commented:
@kb1lqd I finished my first pass through the documentation. Feel free to give it a read-through!

kb1lqd commented on Tutorials/start/Readme.md:
```diff
@@ -34,14 +34,14 @@ Faraday uses a standard Micro-USB connector which means any Micro-USB cable will
 
 ### External MOSFET Connector
-This connector operates in exactly the same fashion as the external power connector P3. However, it is an "output" where the on-board MOSFET simply completes the conduction path when commanded. One wire is VCC which can short to anything referenced to Faraday ground. 
+This connector operates in exactly the same fashion as the external power connector P4. However, it is an "output" where the on-board MOSFET simply completes the conduction path when commanded. One wire is VCC which can short to anything referenced to Faraday ground.
```
File updates all look good!

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/installing-software.md:
```diff
@@ -1,117 +1,89 @@
+## Installing from PyPI
+Just want to run the latest `faraday` software? Open up a command prompt and execute the following commands:
+
+* `pip install faraday`
```
Copy and pasted worked.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/installing-software.md:
```diff
@@ -1,117 +1,89 @@
+2. Navigate to the new folder `cd git/faradayrf`
+3. Clone the lastest master branch `git clone https://github.com/FaradayRF/Faraday-Software.git software`
+4. Navigate to the software `cd software`
+5. Install in pip editable mode with `pip install -e .`
```
Worked!

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/installing-software.md:
```diff
@@ -1,117 +1,89 @@
+
+## Installing in Pip Editable Mode
+If you want to be able to develop with Faraday Software then we suggest cloning our Git Repository as described here.
+
```
ACTION @kb1lqc : You should note that when installing in editable mode that file paths may be awkward and you will need to run faraday programs from the root folder.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/connecting-hardware.md:
````diff
@@ -5,33 +5,33 @@ In order to configure the software with appropriate information we need to plug
+1. In terminal ```ls /dev/ttyUSB*``` to see what USB serial ports are connected.
+2. Note the displayed filenames. This is the device dev path in Linux. If you have other serial devices plugged in you may want to remove Faraday and run the same command to see which file goes away. Faraday would be the file that disapeared.
+
 Faraday will show up as ```/dev/ttyUSB0``` if only one is connected
````
You can also search FOR "FaradayRF" as shown below:
![image](https://user-images.githubusercontent.com/339144/27818913-b8655984-604c-11e7-90a2-e2e1ea0cacb0.png)

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/configuring-proxy.md:
```diff
@@ -1,124 +1,42 @@
+Once installed with `pip` or in editable mode, you can start proxy by simply running `faraday-proxy` from the command line. However it must be properly configured before it will run.
 
-Congratulations, Proxy is now running successfully!
+![Proxy INI](images/proxy.jpg "Proxy command prompt")
```
ACTION @kb1lqc : Image link appears broken.

kb1lqc replied:
Ahh got caught by `*.JPG` files and specifying links to images with `.jpg`. Moved images to lowercase names. ✔️

kb1lqd commented on Tutorials/start/configuring-proxy.md:
```diff
@@ -1,124 +1,42 @@
+`faraday-proxy --callsign kb1lqc --nodeid 2 --port COM23`
 
-> Check your COM port settings and that you did not change `baudrate` in `proxy.ini`
+![Proxy Configuration Example](images/proxy-configuration.jpg "Proxy configuration")
```
ACTION @kb1lqc : Image link appears broken.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/configuring-proxy.md:
```diff
@@ -1,124 +1,42 @@
-### Windows
- 1. [Connect](connecting-hardware.md) ***both*** Faraday radios to your computer via USB and identify their respective COM ports. 
- 2. Open and configure the `proxy.ini` by changing `UNITS` in the [PROXY] section to indicate the quantity of radios connected to the computer. Also add a `[UNITx]` section for each radio.
+![Proxy INI](images/proxy.jpg "Proxy command prompt")
```
ACTION @kb1lqc : Image link appears broken.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/configuring-proxy.md:
```diff
@@ -1,124 +1,42 @@
+Do not be concerned if you see any messges such as that shown below. This is simple a bad packet sent to Proxy over UART which failed CRC checking and was thrown out. This occasionally happens and is OK. With noisy computer grounds and poor quality USB cables one can expect to see more of these messages.
 
-![Device Manager](images/Proxy-INI-Example-Multiple-Units.png "Device Manager")
+![Proxy INI UARTStack Fail](images/proxy-uartstackfail.jpg "Proxy command prompt")
```
ACTION @kb1lqc : Image link appears broken.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/configuring-proxy.md:
````diff
@@ -1,124 +1,42 @@
 
-A proper ```proxy.ini``` configuration file will resemble the example below.
+#### UARTStack Fail Messages
+Do not be concerned if you see any messges such as that shown below. This is simple a bad packet sent to Proxy over UART which failed CRC checking and was thrown out. This occasionally happens and is OK. With noisy computer grounds and poor quality USB cables one can expect to see more of these messages.
````
Action @kb1lqc : Typo "This is simple a bad..." should be "This is simply a bad..."

kb1lqc replied:
Fixed, also found "messges" and fixed that too. ✔️

kb1lqd commented on Tutorials/start/configuring-faraday.md:
```diff
@@ -1,83 +1,69 @@
+![Device Configuration Initialization](images/deviceconfiguration-init.jpg)
+
+## Configuring Device Configuration
+Now we need to tell the `faraday-deviceconfiguration` program where to find Proxy and what to configure the radio with. At this time the resulting command can be long so please be careful. Here are the configuration options:
```
We should create a standard BATCH and BASH script to help automate this for people. Maybe a good simple thing if someone wants to help out!

kb1lqc replied:
Created https://github.com/FaradayRF/Faraday-Software/issues/225 to address this ✔️

kb1lqd commented on Tutorials/start/configuring-faraday.md:
````diff
@@ -1,83 +1,69 @@
+ faraday-deviceconfiguration --proxycallsign kb1lqc --proxynodeid 2 --callsign kb1lqc --nodeid 10 --rftelemetryenable --rfinterval 2 --start
+ ```
+
+At this point the configuration server is running and ready to program our intended data to the radio. If you do need to change a faraday setting and have already told `faraday-deviceconfiguration` where Proxy is then there is no need to reconfigure those values.
````
Action @kb1lqc : Wait, what?

kb1lqc replied:
Fixed wording ✔️

kb1lqd commented on Tutorials/start/configuring-faraday.md:
```diff
@@ -1,83 +1,69 @@
+### Programming CC430 FLASH configuration
+Now the time has come to actually program Faraday. To do so you simply run SimpleConfig which results in Device Configuration sending a POST request to Proxy to program the radio. After waiting five seconds, SimpleConfig will also send a GET request to query the contents of the CC430 FLASH memory helping confirm the intended settings.
+
+* `faraday-simpleconfig`
```
Works.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/configuring-faraday.md:
```diff
@@ -1,83 +1,69 @@
+### Reading the CC430 FLASH Configuration
+We've added a useful feature to SimpleConfig which lets you query Faraday without programming it. Therefore you can see what the current FLASH configuration is. You must have `faraday-proxy` and `faraday-deviceconfiguration` running to perform this action.
+
+* `faraday-simpleconfig --read`
```
Works.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/telemetrystart.md:
```diff
@@ -1,16 +1,36 @@
+
+## Initializing Telemetry
+Just as we've done many times, we need to initialize the  `faraday-telemetry` configuration file. Simply run:
+* `faraday-telemetry --init-config`
```
Works.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/telemetrystart.md:
```diff
@@ -1,16 +1,36 @@
+Just as we've done many times, we need to initialize the  `faraday-telemetry` configuration file. Simply run:
+* `faraday-telemetry --init-config`
+
+![Telemetry Configuration](images/faraday-telemetry-init.jpg)
```
ACTION @kb1lqc : Image link appears broken.

kb1lqc replied:
Fixed ✔️

kb1lqd commented on Tutorials/start/telemetrystart.md:
```diff
@@ -1,16 +1,36 @@
+
+Running this command will start the Telemetry application which should be left running.
+
+![Faraday Telemetry](images/faraday-telemetry.jpg)
```
ACTION @kb1lqc : Image link appears broken.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/telemetrystart.md:
```diff
@@ -1,16 +1,36 @@
-The configuration guide in [Telemetry](../../Applications/Telemetry) documentation sums up how to get the application up and running. We suggest you head over to that documentation to step through configuration. Once setup, play around with the example queries using your web browser located on the bottom of the documentation. Report back here when complete!
+To configure the `faraday-telemetry` application simply tell it what the Proxy callsign and nodeid is as programmed when we configured `faraday-proxy`:
+
+* `faraday-telemetry --callsign kb1lqc --nodeid 2`
```
ACTION @kb1lqc : DOH! I started by using a fresh install of pip (manually deleted the python27 etc folder contents. I got this error because the installation does not install telemetrydb.sql.

![image](https://user-images.githubusercontent.com/339144/27819576-c77ad662-604f-11e7-9a94-ed8e3a945273.png)

![image](https://user-images.githubusercontent.com/339144/27819583-ce00fe44-604f-11e7-940b-3fdc22f941f3.png)

As shown BELOW the GITHUB repo clearly has the correct file:
![image](https://user-images.githubusercontent.com/339144/27819655-11dfc622-6050-11e7-9bfc-b31d3c4e2b89.png)

kb1lqd replied:
Command operated OK after manually copying missing file into etc folder.

kb1lqc replied:
@kb1lqd are you sure you're running the latest pip version? Try `show faraday` and make sure you're using 'faraday 0.0.1014.dev1086'

I created https://github.com/FaradayRF/Faraday-Software/issues/226 to make sure this is fixed but I already fixed the original problem with this where telemetry.sample.ini had 'db.sql' instead of 'telemetrydb.sql' in the schemaname so either you're running old code or there's something else happening.

kb1lqd replied:
![image](https://user-images.githubusercontent.com/339144/27839971-218da3c8-60aa-11e7-8a04-b957d8008075.png)

kb1lqd commented on Tutorials/start/telemetrystart.md:
```diff
@@ -1,16 +1,36 @@
 # Moving Forward
-Now that you are obtaining telemetry from your Faraday radio it is time to command it. Turn on the LED in our [Hello World](hello-world.md) guide!
\ No newline at end of file
+Now that you are obtaining telemetry from your Faraday radio it is time to command it. Turn on the LED in our [Hello World](hello-world.md) guide!
```
ACTION @kb1lqc : You may want to mention where the new telemetry files are stored (in the USERS .faraday/lib folder):

![image](https://user-images.githubusercontent.com/339144/27819774-98b95938-6050-11e7-8772-b0371b72efad.png)

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/hello-world.md:
```diff
@@ -1,68 +1,72 @@
+* Faraday radio connected via USB to the computer
+* Faraday radio configured correctly
+* Proxy configured and running
+* Download [hello-world.py](https://github.com/kb1lqc/Faraday-Software/raw/issue220/Tutorials/start/configuring-proxy.md) to you computer, open in a text editor
```
ACTION @kb1lqc : Uhhh you have the WRONG link here. Sould be `...kb1lqc/Faraday-Software/issue220/Tutorials/start/hello-world.py`

kb1lqd replied:
Do note that when I use the correct file the file operates nominally.

kb1lqc replied:
Aww snap this should NOT point to the issue220 branch anyways. I updated with a link to rawgit file service. This points to the master branch so until this PR is pulled in it will provide the older file currently in master. ✔️

kb1lqd commented on Tutorials/start/hello-world.md:
```diff
@@ -1,68 +1,72 @@
- * Navigate to `C:\faradayrf\faraday-software\Tutorials\start' and run `python hello-world.py`
- 
+* Double-click on hello-world.py and run with python
+* Navigate to `C:\faradayrf\faraday-software\Tutorials\start` and run `python hello-world.py`
```
ACTION @kb1lqc : Wait where is this path from?

kb1lqc replied:
Updated wording for both Windows and Linux instructions. ✔️

kb1lqd commented on Tutorials/start/hello-world.py:
```diff
@@ -15,8 +10,8 @@
 
-callsign = 'REPLACEME'
-node_id = 0
+callsign = 'REPLACEME'  # Callsign string
```
OK looks good.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/configuring-rf-faraday.md:
````diff
@@ -1,88 +1,41 @@
-### Configuration Reference
- Below is a reference of the entire `faraday_config.ini` with an explanation of each setting.
+```
+faraday-deviceconfiguration --proxycallsign kb1lqc --proxynodeid 2 --callsign kb1lqc --nodeid 10 --rftelemetryenabled --rfinterval 2 --start
````
Worked.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/rfplayground.md:
```diff
@@ -11,10 +11,10 @@ The base station, access point, etc is simply a Faraday attached to a computer v
+If you see your local node data displayed in your browser your base station is ready to go!
 
 ## Remote Node
 Next up our Faraday radio which includes GPS on-board will be used for the remote node. This is a Faraday radio with a Serial Number of SN 7 in this example. Already configured for RF transmissions, it needs to be powered and moved at least 10 feet from the base station to prevent desensing of the receiver. Faraday has some strong filters but in-band signals from other 33cm radios may cause probl ...
```
No need for 10 feet if power lower than 20. Maybe like 2-3 feet or more.

kb1lqc replied:
✔️

kb1lqd commented on Tutorials/start/rfplayground.md:
```diff
@@ -89,5 +89,3 @@ Base stations use the APRS "#" symbol representing a "digi" as shown below on [a
 Your Faraday radio is now setup with a callsign and node ID it will use to identify itself even after reboot. We've also installed all necessary programs and configured them to communicate over USB serial ports and over RF using the 33cm ham band. You're ready to go!
```
ACTION @kb1lqc : Line 47 you need to specify WHERE the use can find the telemtry.db file which has changed with recent updates to the Users .faraday folder.

kb1lqc replied:
✔️

kb1lqd reviewed (changes requested):
I left many notes and a number of typo's/broken images/ and info clarifications. Overall the updates were a success and ran nominally but I tried to catch all errors/updates as possible.

kb1lqc committed:
Fixed Git Editable Mode Section

Described filepath issue and clarified installation.

kb1lqc committed:
Removed Redundant Code

Accidentally added redundant pip install code.

kb1lqc committed:
Added grep statement

Search for FaradayRF with grep in Linux.

kb1lqc committed:
Updated Image file extensions

File extensions .JPG seems to be not working when the links point to
*.jpg. Moved files to .jpg extension.

kb1lqc committed:
Fixed Typos

Found two typos and fixed them.

kb1lqc committed:
Updated wording

Fixed confusing wording of settings being static.

kb1lqc committed:
Added New Images

I forgot to add these images while writing the documentation.

kb1lqc committed:
Updated Telemetry Log Location Info

Added a section to explain where the telemetry log data is stored.

kb1lqc committed:
Changed link for hello-world.md

Had completely wrong link in there and now also making it a relative link.

kb1lqc committed:
Updated hello-world.py link

Using rawgit link to provide file from a CDN in the correct MIME type.

kb1lqc committed:
Updated link with new rawgit url

Realized I need to use the development mode link to share the latest
updated version.

kb1lqc committed:
Updated wording for hello-world.py operation

Oddly worded instructions updated.

kb1lqc committed:
Updated wording for RF desensing

Removed 10 foot reference and updated with about 3 feet or more reference
at a power level of 20.

kb1lqc committed:
Added Telemetry Log File Location Info

Copied and pasted log file location text from telemetrystart.md

kb1lqc commented:
@kb1lqd I believe I addressed all issues and those that were not applicable to this PR I created issue tickets for. Anything else?
